lib/api/apiUtils/object/versioning.js

```diff
@@ -133,6 +133,11 @@ function processVersioningState(reqVersionId, mst, vstat, cb) {
             options.versionId = '';
             options.isNull = true;
             options.dataToDelete = mst.objLocation;
+            // if null version exists, clean it up prior to put
+            if (mst.isNull) {
+                delOptions.versionId = mst.versionId;
+                return cb(null, options, null, delOptions);
```

Comment:
Discussed offline with @electrachong: I'm slightly concerned that this incurs a non-atomic operation where potentially old data is deleted without the new data written yet (atomicity issue), or that the old data is lost without being replaced by the new data (put failure), but this seems an orthogonal issue to this PR, the case already exists in other situations.

As this has already been discussed by mail with @ttvi , and since it affects only old data that the user wants removed anyway, I'm ok with it, but I think it should be tracked for further refinements (as I heard it requires additional metadata routes to selectively delete a version).